File: js.v/js/neighbor.js
function whoIsNeighbor(personId) {
    let neighborId;
    for (var i = personId + 1; i != personId; i++) {
        if (i >= peopleC - 1) { i = 0; }
        if (world[i].alive) {
            neighborId = i;
            break;
        }
    }
    return neighborId;
}

function askNeighbor(personId, reason, askedAmount) {
    let neighborId = whoIsNeighbor(personId);
    let maxPossibleHelp = world[neighborId].generousness * 100;
    if (maxPossibleHelp >= askedAmount) {
        world[neighborId][reason] -= askedAmount;
        return askedAmount;
    }
    else{
        return 0;
    }
}

File: py.v/neighbor.py
from main import *

def whoIs(wid):
    neighborwid = None
    i = wid + 1
    while True:
        if i >= peopleC - 1:
            i = 0
        if world[i].alive:
            neighborwid = i
            break
        else:
            i += 1
    return neighborwid

def ask(wid, reason, askedAmount):
    neighborwid = whoIs(wid)
    maxPossibleHelp = world[neighborwid].generosity
    if maxPossibleHelp >= askedAmount:
        world[neighborwid][reason] -= askedAmount
        return askedAmount
    else:
        return False
